clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -117,6 +137,341 @@ class image_mem_wrapper {
   image_mem_wrapper *_sub_wrappers{nullptr};
 };
 
+#ifdef _WIN32
+std::unordered_set<ID3D11Resource *> uniqueD3D11Res;
+#endif
+
+/// The wrapper class of graphics interop memory handle.
+class interop_mem_wrapper {
+public:
+#ifdef _WIN32
+  interop_mem_wrapper(ID3D11Resource *d3d11_res, unsigned reg_flags)
+      : _d3d11_res(d3d11_res), _reg_flags(reg_flags) {
+
+    assert(uniqueD3D11Res.find(_d3d11_res) == uniqueD3D11Res.end() &&
+        "This D3D11 resource is already registered");
+
+    query_res_info(_d3d11_res);
+
+    _win_nt_handle = create_shared_handle(_d3d11_res);
+
+    sycl::ext::oneapi::experimental::resource_win32_handle 
+      ext_mem_win_handle{_win_nt_handle};
+
+    sycl::ext::oneapi::experimental::external_mem_descriptor
+      <sycl::ext::oneapi::experimental::resource_win32_handle> ext_mem_desc{
+        ext_mem_win_handle,
+        sycl::ext::oneapi::experimental::external_mem_handle_type::
+            win32_nt_dx12_resource,
+        _res_size_bytes};
+
+    _interop_mem_handle_ptr =
+      new sycl::ext::oneapi::experimental::interop_mem_handle;
+    *_interop_mem_handle_ptr =
+      sycl::ext::oneapi::experimental::import_external_memory(
+        ext_mem_desc, get_default_queue());
```

Comment:
```suggestion
      new (_interop_mem_handle_ptr) sycl::ext::oneapi::experimental::interop_mem_handle(
      sycl::ext::oneapi::experimental::import_external_memory(
        ext_mem_desc, get_default_queue()));
```
BTW, why using a pointer instead of an object for the handle here?

Reply:
Implemented as a pointer to check if a resource is registered or not (if `_interop_mem_handle_ptr` is NULL or not) during unregistration (destructor).

Reply:
This pointer is initialized in constructor, and I see the check is done at the beginning of the constructor. So I think it cannot be used for the check.

Reply:
I wanted a mechanism to check if the resource was registered or not before releasing its memory in the wrapper destructor ([here](https://github.com/oneapi-src/SYCLomatic/blob/55de4c380edbdf6a04de6d7b13430848b92a8622/clang/runtime/dpct-rt/include/dpct/bindless_images.hpp#L193)).
That's the main motivation for making it a pointer.

The check at the beginning of the constructor is done with the D3D11 resource itself

Reply:
But it is always initialized in the constructor. In destructor, there is no check needed, since it cannot be uninitialized.